chainer/links/connection/n_step_rnn.py

```diff
@@ -0,0 +1,175 @@
+
+    .. seealso::
+        :func:`chainer.functions.n_step_rnn`
+        :func:`chainer.functions.n_step_birnn`
+
+    """
+    def __init__(self, n_layers, in_size, out_size, dropout, use_cudnn,
+                 use_bi_direction):
+        weights = []
+        direction = 2 if use_bi_direction else 1
+        for i in six.moves.range(n_layers):
+            for di in six.moves.range(direction):
+                weight = link.Link()
+                for j in six.moves.range(2):
+                    if i == 0 and j < 1:
+                        w_in = in_size
+                    elif i > 0 and j < 1:
+                        w_in = out_size * direction
+                    else:
+                        w_in = out_size
+                    weight.add_param('w%d' % j, (out_size, w_in))
+                    weight.add_param('b%d' % j, (out_size,))
+                    getattr(weight, 'w%d' % j).data[...] = numpy.random.normal(
+                        0, numpy.sqrt(1. / w_in), (out_size, w_in))
+                    getattr(weight, 'b%d' % j).data[...] = 0
+                weights.append(weight)
+
+        super(NStepRNNBase, self).__init__(*weights)
+
+        self.n_layers = n_layers
+        self.dropout = dropout
+        self.use_cudnn = use_cudnn
+        self.out_size = out_size
+        self.direction = direction
+        self.rnn = rnn.n_step_birnn if use_bi_direction else rnn.n_step_rnn
+
+    def __call__(self, hx, xs, train=True):
+        """Calculate all hidden states and cell states.
+
+        Args:
+            hx (~chainer.Variable or None): Initial hidden states. If ``None``
+                is specified zero-vector is used.
+            xs (list of ~chianer.Variable): List of input sequences.
+                Each element ``xs[i]`` is a :class:`chainer.Variable` holding
+                a sequence.
+        """
+        assert isinstance(xs, (list, tuple))
+        indices = argsort_list_descent(xs)
+
+        xs = permutate_list(xs, indices, inv=False)
+        if hx is None:
+            with cuda.get_device(self._device_id):
+                hx = chainer.Variable(
+                    self.xp.zeros((self.n_layers * self.direction,
+                                   len(xs), self.out_size),
+                                  dtype=xs[0].dtype),
+                    volatile='auto')
+        else:
+            hx = permutate.permutate(hx, indices, axis=1, inv=False)
+
+        trans_x = transpose_sequence.transpose_sequence(xs)
+
+        ws = [[w.w0, w.w1] for w in self]
+        bs = [[w.b0, w.b1] for w in self]
+
+        hy, trans_y = self.rnn(
+            self.n_layers, self.dropout, hx, ws, bs, trans_x,
+            train=train, use_cudnn=self.use_cudnn)
+
+        hy = permutate.permutate(hy, indices, axis=1, inv=True)
+        ys = transpose_sequence.transpose_sequence(trans_y)
+        ys = permutate_list(ys, indices, inv=True)
+
+        return hy, ys
+
+
+class NStepRNN(NStepRNNBase):
```

Comment:
I have modified link code to `activation` arguments and modified test code for `relu` activation.